src/controls/SelectTeamChannelPicker/SelectTeamChannelPicker.tsx

```diff
@@ -0,0 +1,251 @@
+  ISuggestionsItem,
+} from "office-ui-fabric-react/lib/Pickers";
+import { useTeams } from "../../hooks";
+import { ITeamChannel } from "./../../common/model/ITeamChannel";
+import { ISelectTeamChannelPickerProps } from "./ISelectTeamChannelPickerProps";
+import {
+  IconButton,
+  Stack,
+  Text,
+  FontIcon,
+  Label,
+} from "office-ui-fabric-react";
+import { find, pullAllBy } from "lodash";
+import { ISelectTeamChannelPickerState } from "./ISelectTeamChannelPickerState";
+import { useSelectTeamChannelPickerStyles } from "./SelectTeamChannelPickerStyles";
+import { EMembershipType } from "./EMembersipType";
+
+const theme = window.__themeState__.theme;
+
+const initialState: ISelectTeamChannelPickerState = {
+  selectedTeamsChannels: [],
+};
+const getTextFromItem = (item: ITag) => item.name.split(",")[0];
+// Reducer to update state
+const reducer = (
+  state: ISelectTeamChannelPickerState,
+  action: { type: string; payload: any }
+) => {
+  switch (action.type) {
+    case "UPDATE_SELECTEITEM":
+      return { ...state, selectedTeamsChannels: action.payload };
+    default:
+      return state;
+  }
+};
+
+// select Team control
+export const SelectTeamChannelPicker: React.FunctionComponent<ISelectTeamChannelPickerProps> = (
+  props: ISelectTeamChannelPickerProps
+) => {
+  // initialize reducer
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const picker = React.useRef<IBasePicker<ITag>>(null);
+  const { serviceScope } = props.appcontext;
+  const { getTeamChannels } = useTeams(serviceScope);
+  const {
+    renderItemStylesSingle,
+    renderItemStylesMulti,
+    pickerStylesMulti,
+    pickerStylesSingle,
+    renderIconButtonRemoveStyles,
+  } = useSelectTeamChannelPickerStyles(theme);
+  const {
+    onSelectedChannels,
+    selectedChannels,
+    itemLimit,
+    label,
+    styles,
+  } = props;
+
+  const pickerSuggestionsProps: IBasePickerSuggestionsProps = {
+    suggestionsHeaderText: "Suggested Team Channels",
+    noResultsFoundText: "No channels found",
+  };
+
+  /**
+   *  Get Sugested Teams
+   */
+  const useFilterSuggestedTeamsChannels = React.useCallback(
+    async (filterText: string, teamsChannelList: ITag[]): Promise<ITag[]> => {
+      let tags: ITag[] = [];
+      try {
+        const teamsChannels: ITeamChannel[] = await getTeamChannels(
+          props.teamId.toString(),
+          filterText
+        );
+
+        if (teamsChannels?.length) {
+          for (const teamChannel of teamsChannels) {
+            const checkExists = find(teamsChannelList, { key: teamChannel.id });
+            if (checkExists) continue;
+            tags.push({
+              key: teamChannel.id,
+              name: `${teamChannel.displayName},${teamChannel.membershipType},${
+                teamChannel.isFavoriteByDefault ?? "false"
+              }`,
+            });
+          }
+        }
+      } catch (error) {
+        console.log(error);
+      }
+      return tags;
+    },
+    []
```

Comment:
`useCallback` with empty dependency array may lead to issues.
For example, if you use some state variables in the callback, they might always have `undefined` value as the callback was defined before any value has been set to the var.
For this one `props` or `props.teamId` potentially can be `undefined` or empty.